web-ui/controller/git_forge.ml

```diff
@@ -281,4 +308,70 @@ module Make (View : View) = struct
     in
     List.iter (fun (s, m) -> Dream.add_flash_message request s m) flash_messages;
     Dream.redirect request (Fmt.str "/github/%s/%s/commit/%s" org repo hash)
+
+  let show_step_json ~org ~repo ~hash ~variant ci =
+    Backend.ci ci >>= fun ci ->
+    Capability.with_ref (Client.CI.org ci org) @@ fun org_cap ->
+    Capability.with_ref (Client.Org.repo org_cap repo) @@ fun repo_cap ->
+    Capability.with_ref (Client.Repo.commit_of_hash repo_cap hash)
+    @@ fun commit_cap ->
+    let jobs = Client.Commit.jobs commit_cap in
+    Capability.with_ref (Client.Commit.job_of_variant commit_cap variant)
+    @@ fun job_cap ->
+    jobs >>!= fun jobs ->
+    Capability.inc_ref job_cap;
+    let build_created_at =
+      Option.value ~default:0.
+      @@ Option.join
+      @@ Result.to_option
+      @@ Run_time.build_created_at ~build:jobs
```

Comment:
```suggestion
      Run_time.build_created_at ~build:jobs
      |> Result.to_option
      |> Option.join
      |> Option.value ~default:0.
```
IMHO, The composition style with too many functions hides what happens underneath. Feel free to ignore the comment.